src/sinks/util/batch.rs

```diff
@@ -1,47 +1,93 @@
 use serde::{Deserialize, Serialize};
+use snafu::Snafu;
 use std::time::Duration;
 
-#[derive(Clone, Debug, Default, Deserialize, Serialize)]
-pub struct BatchBytesConfig {
+#[derive(Debug, Snafu)]
+pub enum BatchError {
+    #[snafu(display("Cannot configure both `max_bytes` and `max_size`"))]
+    BytesAndSize,
+    #[snafu(display("Cannot configure both `max_events` and `max_size`"))]
+    EventsAndSize,
```

Comment:
```suggestion
    #[snafu(display("Cannot configure both `max_events` and `max_bytes`"))]
    EventsAndBytes,
```

Reply:
No, those two are exactly what they are supposed to be. Due to the backwards compatibility nature of `max_size`, it acts as an alias for either `max_bytes` or `max_events` depending on what `max_size` used to be used for. As such, it is not allowed to configure both at once. It is, on the other hand, possible to configure both `max_bytes` and `max_events`, and that is part of the point of this whole rework.

Now, it might be worth adding a note to that error that `max_size` is now deprecated, but that's different than what you've suggested.